File: lilypad/lilypadmanagerclass.py
class LilypadManager:
    def __init__(self, allLilyPads):
        self.all_lily_pads = #populate list with DB data

    def get_nearest_four(user_locn, radius, pad, free):
        filtered_list = filter_type_and_cost(get_within_radius(user_locn, radius), pad, free);

        if len(filtered_list) >= 4:
            return filtered_list[0:4]
        else:
            return filtered_list

    def get_within_radius(user_locn, radius):
        lilypads_within_radius = []

        for lp in self.all_lily_pads:
            if SphericalGeometry.distance(user_locn, radius) <= radius:
                lilypads_within_radius.append(lp)

        return lilypads_within_radius

    def filter_type_and_cost(lilypads_within_radius, pad, free):
        lilypads_with_type = []

        for lp in lilypads_within_radius:
            if lp.pad == pad and lp.free == free:
                lilypads_with_type.append(lp)

        return lilypads_with_type

File: lilypad/lilypadmanagerclass.cs
public class LilypadManager {
List<Lilypad> allLilyPads;
public LilypadManager() {
  allLilyPads = new ArrayList<Lilypad>();
  //populate list with data from DB
}

public List<Lilypad> getNearestFour(LatLon userLocn, double radius, bool pad, bool free) {
  List<Lilypad> filteredList = filterTypeAndCost(getWithinRadius(userLocn, radius), bool pad, bool free);
  if (filteredList.count >= 4) {
    return filteredList.sublist(0,4);
  } else {
    return filteredList;
  }
}

public List<Lilypad> getWithinRadius(LatLon userLocn, double radius) {
  List<Lilypad> lilypadsWithinRadius = new ArrayList<LilyPad>();
  //Jess' code to query database for getting all LilyPad records
  //put into allLilyPads
  for (Lilypad lp : allLilyPads) {
    double distance = SphericalGeometry.distanceBetween(userLocn, lp);
    if (lp <= userLocn) {
      lilypadsWithinRadius.add(lp);
    }
  }
    return lilypadsWithinRadius;
  }

public List<Lilypad> filterTypeAndCost(List<Lilypad> lilypadsWithinRadius, bool pad, bool free) {
  List<Lilypad> lilypadsWithType = new ArrayList<LilyPad>();

  for (Lilypad lp : lilypadsWithinRadius) {
    if (lp.Pad == pad && lp.free == free) {
      lilypadsWithType.add(lp);
    }
  }
  return lilypadsWithType;
}

}
